mdm/hibernate-core/src/main/java/com/gentics/mesh/hibernate/data/dao/DaoHelper.java

```diff
@@ -659,13 +665,19 @@ <S extends SortingParameters> Pair<NativeJoin, Map<String, SortOrder>> mapSortin
 				} else {
 					String[] localKeyParts = keyParts;
 					if (i > 0) {
-						localKeyParts = Arrays.copyOfRange(localKeyParts, i, localKeyParts.length); // Collections.singletonList(IntStream.range(i, keyParts.length).mapToObj(start -> keyParts[start]).collect(Collectors.joining("."))).toArray(new String[keyParts.length-i]);
+						localKeyParts = Arrays.copyOfRange(localKeyParts, i, localKeyParts.length); 
 					}
 					boolean noAlias = StringUtils.isBlank(alias);
 					Pair<String, String> pair = makeSortJoins(alias, localKeyParts, mapper, ansiJoin, maybeContainerType, maybeBranch, baseJoin, otherJoins);
+					boolean noDomainColumn = (keyParts.length < 2) && domainColumns.map(columns -> columns.stream().allMatch(column -> !column.equalsIgnoreCase(pair.getRight()))).orElse(false);
```

Comment:
We check against domain columns only for the 1st level entities, otherwise it is too cumbersome to look through all the possible solutions.